tests/test_cookiejar.py

```diff
@@ -1205,3 +1206,218 @@ async def test_filter_cookies_does_not_leak_memory() -> None:
     for key, morsels in jar._morsel_cache.items():
         assert key in jar._cookies, f"Orphaned morsel cache entry for {key}"
         assert len(morsels) > 0, f"Empty morsel cache entry found for {key}"
+
+
+def test_update_cookies_from_headers() -> None:
+    """Test update_cookies_from_headers method."""
+    jar: CookieJar = CookieJar()
+    url: URL = URL("http://example.com/path")
+
+    # Test with simple cookies
+    headers: List[str] = [
```

Comment:
```suggestion
    headers = [
```
mypy can already work this out